lib/ts/recipe/passwordless/components/features/linkClickedScreen/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { getQueryParams, getRedirectToPathFromURL, getURLHash } from "../../../../../utils";
+import FeatureWrapper from "../../../../../components/featureWrapper";
+import { StyleProvider } from "../../../../../styles/styleContext";
+import { defaultPalette } from "../../../../../styles/styles";
+import { getStyles } from "../../themes/styles";
+import {} from "../../../types";
+import { LinkClickedScreen } from "../../themes/linkClickedScreen";
+import Recipe from "../../../recipe";
+import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
+
+type PropType = FeatureBaseProps & { recipe: Recipe };
+
+class SignInAndUpCallback extends PureComponent<PropType, unknown> {
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    componentDidMount = async (): Promise<void> => {
+        try {
+            const preAuthSessionId = getQueryParams("preAuthSessionId");
+            const linkCode = getURLHash();
+
+            if (preAuthSessionId === null || preAuthSessionId.length === 0 || linkCode.length === 0) {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "signin",
+                });
+            }
+
+            const response = await this.props.recipe.recipeImpl.consumeCode({
+                preAuthSessionId,
+                linkCode,
+                config: this.props.recipe.config,
+            });
+
+            if (response.status === "GENERAL_ERROR") {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "custom",
+                    message: response.message,
+                });
+            }
+
+            if (response.status === "RESTART_FLOW_ERROR") {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "signin",
+                });
+            }
+            if (response.status === "OK") {
+                this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+                return this.props.recipe.redirect(
+                    { action: "SUCCESS", isNewUser: response.createdUser, redirectToPath: getRedirectToPathFromURL() },
+                    this.props.history
+                );
+            }
+        } catch (err) {
+            return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                error: "signin",
+            });
+        }
+    };
+
+    render = (): JSX.Element => {
+        const componentOverrides = this.props.recipe.config.override.components;
+
+        const linkClickedScreen = this.props.recipe.config.linkClickedScreen;
+
+        const props = {
+            recipeImplementation: this.props.recipe.recipeImpl,
```

Comment:
You want to get a modified version of this that:
- on consumeCode, it clears the login attempt state if successful?
- on restart flow error, it clears the login attempt state if successful?
- anything else?

Reply:
I don't think the restart flow error should clear because it would block you from entering the user input code in the combined flow, which may be a use-case if the user noticed copying the link didn't work and tries to just type it in.

Reply:
However, if the user mistyped the user input code incorrectly a bunch of times, that would result in the flow to restart (which would prevent the magic link from working even if used correctly - is that correct?)

Reply:
That's correct. In this case the user input code part of the flow should do the cleanup. Thinking about this in more general terms, we shouldn't really rely on expired/invalid devices staying in DB, because the last allowed attempt will delete the device as well, so other attempts to consume either the link or the user input code would only see that the device doesn't exists.
If this becomes a problem in the future we could have a separate endpoint where the UI could check if the login attempt is still valid (polling by deviceIdHash)